query-sql/src/lib.rs

```diff
@@ -343,7 +345,15 @@ impl QueryFragment for TableOrSubquery {
     fn push_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         use self::TableOrSubquery::*;
         match self {
-            &Table(ref sa) => source_alias_push_sql(out, sa)
+            &Table(ref sa) => source_alias_push_sql(out, sa),
+            &Union(ref subqueries, ref table_alias) => {
+                out.push_sql("(");
+                interpose!(subquery, subqueries,
+                           { subquery.push_sql(out)? },
+                           { out.push_sql(" UNION ") });
```

Comment:
Again -- `UNION DISTINCT`?  Or are you confident that you'll be able to handle such considerations at the root, and there's never a performance advantage to pushing the `DISTINCT` down to the leaves?